### ISSUE TITLE
Don't buffer stream inside of toString()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOL#259]
`Seq` currently buffers any stream when calling `toString()` because this allows for a nice display in debuggers without wasting the stream / modifying program behaviour.

However, buffering the stream has side effects, mainly also when debugging, breakpoints inside of a stream's lambda transformations might no longer be triggered because the stream is evaluated / consumed on `toString()`, no longer in the real code. See:
https://youtrack.jetbrains.com/issue/IDEA-160867

**Comment by tcn** [comment]
Streams are aimed to large(r) data sets (i.e. toString() is pointless anyways) and on purpose composable and lazily evaluated. Consider debugging through this:
```
final Seq<Integer> cache1 = Seq.of(1, 2, 3);	// and a whole lot more
final Seq<Integer> cache2 = cache1.filter(whatever);	
final Optional<Integer> r = cache2.sum();	// *this* is expected to land on the heap
```

And the current `toString()` doesn't even comma-separate the values :)

**Comment by lukaseder** [comment]
Personally, I don't buy the "large(r)" data set argument. Most people will use streams as glorified collections, i.e. stuff with 1000 elements in it, where the stream source is in memory already.

Anyway, I see your point :)

**Comment by tcn** [comment]
That's why `Seq.parallel()` returns `this`, I suppose :)

**Comment by lukaseder** [comment]
No, why? `Seq.parallel()` returns `this`, because it makes no sense to have a parallel sequential stream

**Comment by tcn** [comment]
https://blog.jetbrains.com/idea/2016/09/intellij-idea-2016-3-eap-helps-migrate-to-java-8s-stream-apis/ ok, you were probably right :)

**Comment by lukaseder** [comment]
Wow, that's impressive. But what do you mean by "I was right"? With what?

**Comment by tcn** [comment]
Uhm, well, that "most" people will use Stream/Seq as "glorified collections"; now that IntelliJ has this particular feature.

**Comment by lukaseder** [comment]
Aha, I thought it was referring to the `parallel()` comment. Yes indeed, that's the assumption here. In fact, that's the whole reason for being of this library. People wanted cooler collections (as in Scala). Not infinite parallel Streams. Only Brian Goetz wanted that, and 1-2 of his peers ;) (Scala has those features too, but hardly anyone uses them)

**Comment by tcn** [comment]
Hardly anyone uses DB cursors.
